### PR TITLE
RET-3462 - Tackle the CognitiveComplexity and CyclomaticComplexity

### PR DESCRIPTION
**Pull request by aaronjduffy**
https://tools.hmcts.net/jira/browse/RET-3462
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java [hunk 1]
```diff
@@ -199,41 +200,43 @@ public Optional<String> validateMaximumSize(CaseData caseData) {
...
-                        .noneMatch(r -> r.getValue().getDynamicRespRepName().getValue().getLabel()
-                            .equals(tempCollCurrentName))) {
-                        repCollection.get(index).getValue().setRespRepName(tempCollCurrentName);
-                        updatedRepList.add(repCollection.get(index));
-                    }
-                } else {
-                    errors.add(RESP_REP_NAME_MISMATCH_ERROR_MESSAGE + " - " + tempCollCurrentName);
-                    return errors;
-                }
+        List<RepresentedTypeRItem> repCollection = caseData.getRepCollection();
```
Comment (furnivall) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java:203]: This is still a pretty huge method - any chance you could break it up into smaller chunks? Same goes for others. Happy to put a comment on the ticket so you can bump up the points

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReport.java [hunk 2]
```diff
@@ -81,18 +81,21 @@ private boolean hasMultipleRespondents(RespondentsReportCaseData caseData) {
     }
 
     private boolean isRepresentativeRepresentingMoreThanOneRespondent(String rep, RespondentsReportCaseData caseData) {
```
Comment (furnivall) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReport.java:83]: This could also be broken up a bit

**Review by furnivall — CHANGES_REQUESTED** [review]

**Review by furnivall — DISMISSED** [review]
Thanks for making those changes, and sorry for the extra hassle!

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 3]
```diff
@@ -185,7 +186,7 @@ public void dateToCurrentPosition(CaseData caseData) {
     }
 
     public void setEt3ResponseDueDate(CaseData caseData) {
-        if (!isNullOrEmpty(caseData.getClaimServedDate()) && isNullOrEmpty(caseData.getEt3DueDate())) {
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:188]: is this removal intentional? I think this logic was put in place so that et3DueDate would only get set if a value wasn't present. Removing this means that the value will potentially be updated every single time

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReport.java [hunk 4]
```diff
@@ -81,19 +81,32 @@ private boolean hasMultipleRespondents(RespondentsReportCaseData caseData) {
...
+
+    private int countMatchingRespondents(String respRepName, RespondentsReportCaseData caseData) {
+        int count = 0;
+        for (RespondentSumTypeItem respItem : caseData.getRespondentCollection()) {
+            if (respItem.getValue().getRespondentName().equals(respRepName)) {
+                count++;
+            }
+        }
+        return count;
     }
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReport.java:110]: Using a stream here might make it cleaner, no need to declare `count` as 0 
```suggestion
private int countMatchingRespondents(String respRepName, RespondentsReportCaseData caseData) {
    return (int) caseData.getRespondentCollection().stream()
            .filter(respItem -> respItem.getValue().getRespondentName().equals(respRepName))
            .count();
}
```

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReport.java [hunk 5]
```diff
@@ -81,19 +81,32 @@ private boolean hasMultipleRespondents(RespondentsReportCaseData caseData) {
...
-                    }
-                }
+        for (RepresentedTypeRItem repItem : caseData.getRepCollection()) {
+            if (repItem.getValue().getNameOfRepresentative().equals(rep)) {
+                count += countMatchingRespondents(repItem.getValue().getRespRepName(), caseData);
             }
         }
-        return count > 1;
+        return count;
+    }
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReport.java:100]: Github won't let me make a suggestion here, but using streams might make this tidier

```
private int countRepresentedRespondents(String rep, RespondentsReportCaseData caseData) {
    return caseData.getRepCollection().stream()
            .filter(repItem -> repItem.getValue().getNameOfRepresentative().equals(rep))
            .mapToInt(repItem -> countMatchingRespondents(repItem.getValue().getRespRepName(), caseData))
            .sum();
}
```

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerServiceTest.java [hunk 6]
```diff
@@ -867,34 +867,6 @@ void setEt3ResponseDueDate_Serving_Date_Plus_28() {
         assertEquals(expectedEt3DueDate, caseData.getEt3DueDate());
     }
 
-    @Test
-    void setNextListedDate() {
```
Comment (JackReevies) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerServiceTest.java:871]: Why remove this test? It looks like you've not touched setNextListedDate in the original code

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 7]
```diff
@@ -268,46 +269,55 @@ public CaseData continuingRespondent(CCDRequest ccdRequest) {
...
+            }
+
+            for (DateListedTypeItem dateListedTypeItem : hearingDateCollection) {
+                DateListedType dateListedType = dateListedTypeItem.getValue();
+                if (dateListedType.getHearingStatus() == null) {
+                    initializeHearingStatus(dateListedType);
                 }
+                populateHearingVenueFromHearingLevelToDayLevel(dateListedType, hearingType, caseTypeId);
             }
         }
```
[the remark this answers is not in the document]
Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java]: Sorry, my mistake. This doesn't work, `hearingType` goes out of scope and the fix with streams looks more complex than the original

**Review by JackReevies — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
